03.bowling/bowling.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+scores = ARGV[0].split(',')
+shots = scores.map { |score| score == 'X' ? 10 : score.to_i }
+
+frames = []
+current_frame = []
+
+shots.each do |shot|
+  current_frame << shot
+  if current_frame[0] == 10 || current_frame.size == 2
+    frames << current_frame
+    current_frame = []
+  end
+end
+
+frames << current_frame unless current_frame.empty?
+
+total_score = 0
+
+frames.take(10).each_with_index do |frame, i|
+  next_frame = frames[i + 1]
+  frame_after_next = frames[i + 2]
+
+  if frame[0] == 10
+    total_score += next_frame.size == 1 ? 10 + next_frame.sum + frame_after_next[0] : 10 + next_frame.take(2).sum
```

Comment:
```suggestion
    following_shots = next_frame + frame_after_next
    total_score += 10 + following_shots.take(2).sum
```

何も考えずに次の2投を取ってくる、というロジックに書き直すこともできます。

Reply:
`total_score += next_frame.size == 1 ? 10 + next_frame.sum + frame_after_next[0] : 10 + next_frame.take(2).sum`を以下のコードのように、何も考えずに次の2投を取ってくる、というロジックに書き直しました。
- `following_shots = next_frame + frame_after_next`
- `total_score += 10 + following_shots.take(2).sum`